src/Xamarin.Android.Build.Tasks/Tasks/CreateDynamicFeatureManifest.cs

```diff
@@ -0,0 +1,140 @@
+
+		public string FeatureTitleResource { get; set; }
+
+		public string MinSdkVersion { get; set; }
+
+		public string TargetSdkVersion { get; set; }
+
+		public bool IsFeatureSplit { get; set; } = false;
+		public bool IsInstant { get; set; } = false;
+		public bool HasCode { get; set; } = false;
+
+		public override bool RunTask ()
+		{
+			XDocument doc = new XDocument ();
+			switch (FeatureType) {
+				case "AssetPack":
+					GenerateAssetPackManifest (doc);
+					break;
+				default:
+					GenerateFeatureManifest (doc);
+					break;
+			}
+			doc.SaveIfChanged (OutputFile.ItemSpec);
+			return !Log.HasLoggedErrors;
+		}
+
+		void GenerateFeatureManifest (XDocument doc) {
+			XAttribute featureTitleResource = null;
+			if (!string.IsNullOrEmpty (FeatureTitleResource))
+				featureTitleResource = new XAttribute (distNS + "title", FeatureTitleResource);
+			XElement usesSdk = new XElement ("uses-sdk");
+			if (!string.IsNullOrEmpty (MinSdkVersion))
+				usesSdk.Add (new XAttribute (androidNS + "minSdkVersion", MinSdkVersion));
+			if (!string.IsNullOrEmpty (MinSdkVersion))
+				usesSdk.Add (new XAttribute (androidNS + "targetSdkVersion", TargetSdkVersion));
+			doc.Add (new XElement ("manifest",
+					new XAttribute(XNamespace.Xmlns + "android", androidNS),
+					new XAttribute(XNamespace.Xmlns + "tools", toolsNS),
+					new XAttribute(XNamespace.Xmlns + "dist", distNS),
+					new XAttribute (androidNS + "versionCode", "1"),
+					new XAttribute (androidNS + "versionName", "1.0"),
+					new XAttribute ("package", PackageName),
+					new XAttribute ("featureSplit", FeatureSplitName),
+					new XAttribute (androidNS + "isFeatureSplit", IsFeatureSplit),
+					new XElement (distNS + "module",
+						featureTitleResource,
+						new XAttribute (distNS + "instant", IsInstant),
+						new XElement (distNS + "delivery",
+							GetDistribution ()
+						),
+						new XElement (distNS + "fusing",
+							new XAttribute (distNS + "include", true)
+						)
+					),
+					usesSdk,
+					new XElement ("application",
+						new XAttribute (androidNS + "hasCode", HasCode),
+						new XAttribute (toolsNS + "replace", "android:hasCode")
+					)
+				)
+			);
+		}
+
+		void GenerateAssetPackManifest (XDocument doc) {
+			doc.Add (new XElement ("manifest",
+					new XAttribute (XNamespace.Xmlns + "android", androidNS),
+					new XAttribute (XNamespace.Xmlns + "tools", toolsNS),
+					new XAttribute (XNamespace.Xmlns + "dist", distNS),
+					new XAttribute ("package", PackageName),
+					new XAttribute ("split", FeatureSplitName),
+					new XElement (distNS + "module",
+						new XAttribute (distNS + "type", "asset-pack"),
+						new XElement (distNS + "delivery",
+							GetDistribution ()
+						),
+						new XElement (distNS + "fusing",
+							new XAttribute (distNS + "include", true)
+						)
+					)
+				)
+			);
+		}
+
+		XElement GetDistribution ()
+		{
+			XElement distribution;
+			switch (FeatureDeliveryType)
```

Comment:
Should these `case` values be case-sensitive?  Or should we have `switch (FeatureDeliveryType?.ToLowerInvariant())` and `case "ondemand"`?

Reply:
Considering that these are user-provided values, we should `.ToLowerInvariant()` things.

We should *also* warn on an unsupported value e.g. if someone does:

```xml
<AndroidAsset DeliveryType="lolDoesNotExist" />
```

then we should warn that "`%(AndroidAsset.DeliveryType)` has has unsupported value "lolDoesNotExist".  And figure out how to provide appropriate filename & line number info?

Reply:
the case sensitive things I can do. The warnings I can also do, providing file and line numbers might be along the lines of "can I have a pony" 🤣 but I'll see what I can do.

Reply:
Yup, line numbers are tricky, especially with autoimports happening. Best we can do is print the "asset" file which had the attribute added to it.